### PR TITLE
Model loading

### PR DESCRIPTION
**Pull request by tigranfah**
A few of key things are implemented

* slight modification of the json processing code, the json str is explicitly parsed under a. try except block, because the previous implementation raise an Exception connected to not being able to parse the json string.
* tune the config files for the maximization of training throughput and reproduction training for chemlactica 125m and 1.3b trainings, add gradient clipping after each forward pass to prevent overflows
* add a debug config for chemlactica model for debugging purposes

### CODE REVIEW

File: torchtitan/logging.py [hunk 1]
```diff
@@ -27,6 +27,9 @@ def init_logger(log_level):
     # suppress verbose torch.profiler logging
     os.environ["KINETO_LOG_LEVEL"] = "5"
 
+    # enable dataloading logging for logging the type of dataloading used
+    enable_dataloader_logging()
```
Comment (philippguevorguian) [inline torchtitan/logging.py:31]: Preferable to propagate log_level to this so we can easily get huggingface dataloader debug logs if needed.

File: torchtitan/metrics.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def build_metric_logger(
     """
     dump_dir = job_config.job.dump_folder
     aim_config = job_config.metrics
-    save_aim_folder = aim_config.save_aim_folder
+    save_aim_folder = os.path.join(job_config.job.dump_folder, aim_config.save_aim_folder)
```
Comment (philippguevorguian) [inline torchtitan/metrics.py:130]: I think the aim folder should be separate from the `dump_folder`. What is the proposed workflow for this method? Is the idea that all runs should then be copied to the central aim repo afterwards?

File: train.py [hunk 3]
```diff
@@ -284,12 +285,14 @@ def loss_fn(pred, labels):
                     # need to free to before bwd to avoid peaking memory
                     del pred
                     loss.backward()
+                
+                for m in model_parts:
```
Comment (philippguevorguian) [inline train.py:289]: No need for changes, but for documentation purposes, when did we observe overflow here?

**Review by philippguevorguian — CHANGES_REQUESTED** [review]

File: torchtitan/metrics.py (hunk 2 above)
Comment (tigranfah) [reply torchtitan/metrics.py:130]: in my mind dump_folder is something like `/nfs/h100/chem` and save_aim_folder `all_aim`, so that all experiments end up there. afterword the main runs are selected and moved to some other aim repo which will need to define.

File: train.py (hunk 3 above)
Comment (tigranfah) [reply train.py:289]: This is a problem, especially at the beginning of the training when the grades are big. Don't recall the exact instance but as soon as you remove it everything is smooth.

File: torchtitan/metrics.py (hunk 2 above)
Comment (philippguevorguian) [reply torchtitan/metrics.py:130]: OK, sounds good.

File: torchtitan/logging.py (hunk 1 above)
Comment (philippguevorguian) [reply torchtitan/logging.py:31]: @tigranfah let's just get this to correspond to our log level, rest lgtm

**Review by philippguevorguian — APPROVED** [review]
lgtm
